### DISCUSSION TITLE
InnoDB 어댑티브 해시 인덱스?

### DISCUSSION THREAD

**Discussion by CODe5753** [Well-Founded-Confidence/RealMySQL#10] [Q&A]
인덱스에도 많은 기능이 있고 종류도 있을텐데요,
Adaptive Hash Index는 처음 들어봤는데, 어떤 걸까요?

**Comment by menthamin** [comment]
1. 인덱스를 해시 테이블에 저장
2. 해시 테이블에 참조하는 주소로 접근 O(1)
3. 자주 사용하는 쿼리 등을 미리 저장해두고 바로 접근

MySQL 엔진이 자체적으로
Adaptive Hash Index에 자주 사용하는 쿼리 등을 저장한다.

**Comment by menthamin** [comment]
key_buffer_size 변수로 Adaptive Hash Index 사이즈를 조정할 수 있다.
InnoDB에서는 사용하지 않음

Adaptive Hash Index: MyISAM에서만 사용

InnoDB: B-tree 인덱스는 레인지스캔에 유리하다. 하단에 리프끼리 링크드리스트로 연결
